op-deployer/pkg/deployer/flags.go

```diff
@@ -117,6 +127,22 @@ var (
 			"intent-config-type",
 		},
 	}
+	EtherscanAPIKeyFlag = &cli.StringFlag{
+		Name:     EtherscanAPIKeyFlagName,
+		Usage:    "etherscan API key for contract verification.",
+		EnvVars:  PrefixEnvVar("ETHERSCAN_API_KEY"),
+		Required: true,
+	}
+	ContractBundleFlag = &cli.StringFlag{
+		Name:    ContractBundleFlagName,
+		Usage:   "contract bundle/grouping (superchain|implementations|opchain)",
+		EnvVars: PrefixEnvVar("CONTRACT_BUNDLE"),
+	}
+	ContractNameFlag = &cli.StringFlag{
```

Comment:
If someone is deploying implementations using a bootstrap command (like we do), then the contracts won't exist in the state file. I recommend instead consuming a JSON of contract names and addresses, then verifying based on the contract name rather than passing in the bundle directly.

Reply:
Also trying to think about how we can accommodate both scenarios:

1. read from a generic json input file
2. read from the state.json file (since this is where opchain contract data is stored)

Do you think we should ignore [2] completely for now and not worry about opchain contract verification for now?

Reply:
For now I am making the assumption `op-deployer verify` will always read from a simple input .json file of the format output by bootstrap commands to get the contracts names + addresses. So I removed the ability to read from the state.json for now. That helped simplify the code nicely. Done here: https://github.com/ethereum-optimism/optimism/pull/14633/commits/8405a9e6b9f4ab07ae40d672625fe2a872641d1b